posthog/clickhouse/property_groups.py

```diff
@@ -0,0 +1,75 @@
+from collections.abc import Iterable, MutableMapping
+from dataclasses import dataclass
+
+from posthog import settings
+
+
+@dataclass
+class PropertyGroupDefinition:
+    filter_expression: str
+    codec: str = "ZSTD(1)"
+
+
+class PropertyGroupManager:
+    def __init__(self, cluster: str, table: str, source_column: str) -> None:
+        self.__cluster = cluster
+        self.__table = table
+        self.__source_column = source_column
+        self.__groups: MutableMapping[str, PropertyGroupDefinition] = {}
+
+    def register(self, name: str, definition: PropertyGroupDefinition) -> None:
+        assert name not in self.__groups, "property group names can only be used once"
+        self.__groups[name] = definition
+
+    def __get_map_expression(self, definition: PropertyGroupDefinition) -> str:
+        return f"mapSort(mapFilter((key, _) -> {definition.filter_expression}, CAST(JSONExtractKeysAndValues({self.__source_column}, 'String'), 'Map(String, String)')))"
+
+    def get_alter_table_statements(self, name: str) -> Iterable[str]:
```

Comment:
```suggestion
    def get_alter_create_statements(self, name: str) -> Iterable[str]:
```

Just making this a bit more clear that it's alter creates